### PR TITLE
base: update to OpenWrt 24.10

### PR DESCRIPTION
**Pull request by blocktrron**
This PR switches Gluon to the latest OpenWrt 24.10 release. 

With this release, support or 2 devices is dropped due to them not fulfilling the minimum flash space requirements anymore.

### CODE REVIEW

**Comment by blocktrron** [comment]
Question is whether we want to do a Wrap-Up release on 23.05 with everything up to this point and switch afterwards. We did this in the past and i think of this as a good way to create a stable release EOY.

**Comment by rotanid** [comment]
> Question is whether we want to do a Wrap-Up release on 23.05 with everything up to this point and switch afterwards. We did this in the past and i think of this as a good way to create a stable release EOY.

thank you for your work and i would agree with doing a Wrap-up v2024.1 Release.

**Comment by rotanid** [comment]
removing the Edgerouter X might be obsolete after that backport:
https://github.com/openwrt/openwrt/pull/16893

**Comment by blocktrron** [comment]
I don't have said device, can't say anything about this backport.

File: contrib/docker/Dockerfile [hunk 1]
```diff
@@ -9,7 +9,10 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     file \
     git \
     python3 \
+    python3-dev \
     python3-distutils \
+    python3-pyelftools \
+    python3-setuptools \
```
Comment (neocturne) [inline contrib/docker/Dockerfile:15]: This adds a lot more packages than are mentioned in the commit message.

File: targets/mediatek-mt7622 [hunk 2]
```diff
@@ -13,9 +13,3 @@ device('ubiquiti-unifi-6-lr-v1', 'ubnt_unifi-6-lr-v1', {
 	manifest_aliases = {'ubiquiti-unifi-6-lr'}, -- Upgrade from OpenWrt 22.03
 })
 
-
--- Xiaomi
-
-device('xiaomi-redmi-router-ax6s', 'xiaomi_redmi-router-ax6s', {
```
Comment (neocturne) [inline targets/mediatek-mt7622:19]: Doc update?

File: targets/ramips-mt7621 [hunk 3]
```diff
@@ -166,17 +166,3 @@ device('zyxel-wsm20', 'zyxel_wsm20', {
...
--- Devices without WLAN
-
--- Ubiquiti
-
-device('ubiquiti-edgerouter-x', 'ubnt_edgerouter-x', {
-	factory = false,
-	packages = {'-hostapd-mini'},
-})
-
-device('ubiquiti-edgerouter-x-sfp', 'ubnt_edgerouter-x-sfp', {
```
Comment (neocturne) [inline targets/ramips-mt7621:179]: Doc updates?

File: scripts/copy_output.lua [hunk 4]
```diff
@@ -92,5 +92,6 @@ if (env.GLUON_DEVICES or '') == '' then
 	lib.exec {'rm', '-f', dest_dir('\0')..'/\0'}
 	lib.exec({'rmdir', '-p', dest_dir('\0')}, true, '2>/dev/null')
 	mkdir(dest_dir(package_prefix))
+	lib.exec {'rm', '-rf', 'openwrt/bin/targets/'..bindir..'/packages/tmp\0'}
```
Comment (neocturne) [inline scripts/copy_output.lua:95]: The `\0` (`*` wildcard) seems unnecessary, or can the directory name vary?
```suggestion
	lib.exec {'rm', '-rf', 'openwrt/bin/targets/'..bindir..'/packages/tmp'}
```

**Review by neocturne — CHANGES_REQUESTED** [review]

**Comment by autinerd** [comment]
We from Freifunk Stuttgart are currently creating an experimental build based on this PR, but with OpenWrt 24.10-rc2 and re-added EdgeRouter X support, we will try to test if the backport works.

**Pull request merged**